### PR TITLE
DOC: Fix RangeIndex and other docstrings for missing period in summary

### PR DESCRIPTION
**Pull request by bhuvanakundumani**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

@datapythonista 

I added the period to the following cases:
pandas.Categorical.dtype: Summary does not end with a period
pandas.merge_ordered: Summary does not end with a period
pandas.bdate_range: Summary does not end with a period
pandas.period_range: Summary does not end with a period
pandas.timedelta_range: Summary does not end with a period
pandas.interval_range: Summary does not end with a period
pandas.util.hash_pandas_object: Summary does not end with a period
pandas.Grouper: Summary does not end with a period
pandas.Index.memory_usage: Summary does not end with a period
pandas.Index.fillna: Summary does not end with a period
pandas.Index.dropna: Summary does not end with a period
pandas.RangeIndex.start: Summary does not end with a period
pandas.RangeIndex.stop: Summary does not end with a period
pandas.RangeIndex.step: Summary does not end with a period

### CODE REVIEW

File: pandas/core/indexes/range.py [hunk 1]
```diff
@@ -236,15 +236,15 @@ def _format_with_header(self, header, na_rep="NaN", **kwargs):
     @cache_readonly
     def start(self):
         """
-        The value of the `start` parameter (``0`` if this was not supplied)
+        The value of the `start` parameter.(``0`` if this was not supplied)
```
Comment (datapythonista) [inline pandas/core/indexes/range.py]: the period should be at the end

File: pandas/core/indexes/range.py [hunk 2]
```diff
@@ -236,15 +236,15 @@ def _format_with_header(self, header, na_rep="NaN", **kwargs):
...
         """
         # GH 25710
         return self._range.start
 
     @property
     def _start(self):
         """
-        The value of the `start` parameter (``0`` if this was not supplied)
+        The value of the `start` parameter.(``0`` if this was not supplied)
```
Comment (datapythonista) [inline pandas/core/indexes/range.py]: same

File: pandas/core/indexes/range.py [hunk 3]
```diff
@@ -282,15 +282,15 @@ def _stop(self):
     @cache_readonly
     def step(self):
         """
-        The value of the `step` parameter (``1`` if this was not supplied)
+        The value of the `step` parameter.(``1`` if this was not supplied)
```
Comment (datapythonista) [inline pandas/core/indexes/range.py]: same

File: pandas/core/indexes/range.py [hunk 4]
```diff
@@ -282,15 +282,15 @@ def _stop(self):
...
         """
         # GH 25710
         return self._range.step
 
     @property
     def _step(self):
         """
-        The value of the `step` parameter (``1`` if this was not supplied)
+        The value of the `step` parameter.(``1`` if this was not supplied)
```
Comment (datapythonista) [inline pandas/core/indexes/range.py]: same

File: pandas/core/reshape/merge.py [hunk 5]
```diff
@@ -177,7 +177,7 @@ def merge_ordered(
 ):
     """
     Perform merge with optional filling/interpolation designed for ordered
-    data like time series data. Optionally perform group-wise merge (see
+    data like time series data. Optionally perform group-wise merge.(see
```
Comment (datapythonista) [inline pandas/core/reshape/merge.py]: same

**Review by datapythonista — COMMENTED** [review]
Thanks for working on this, added some comments, but looks good.

**Comment by bhuvanakundumani** [comment]
@datapythonista - have made the requested changes. Thanks

File: pandas/core/indexes/numeric.py [hunk 6]
```diff
@@ -168,7 +168,7 @@ def _union(self, other, sort):
 ] = """
     Immutable ndarray implementing an ordered, sliceable set. The basic object
     storing axis labels for all pandas objects. %(klass)s is a special case
-    of `Index` with purely %(ltype)s labels. %(extra)s
+    of `Index` with purely %(ltype)s labels.
```
Comment (datapythonista) [inline pandas/core/indexes/numeric.py]: Can you revert this one for now. It's a bit more complex, we'll take care of it in a separate PR.

File: pandas/core/indexes/range.py [hunk 7]
```diff
@@ -236,15 +236,15 @@ def _format_with_header(self, header, na_rep="NaN", **kwargs):
     @cache_readonly
     def start(self):
         """
-        The value of the `start` parameter (``0`` if this was not supplied)
+        The value of the `start` parameter(``0`` if this was not supplied).
```
Comment (datapythonista) [inline pandas/core/indexes/range.py]: can you restore the space please

File: pandas/core/indexes/range.py [hunk 8]
```diff
@@ -236,15 +236,15 @@ def _format_with_header(self, header, na_rep="NaN", **kwargs):
...
         """
         # GH 25710
         return self._range.start
 
     @property
     def _start(self):
         """
-        The value of the `start` parameter (``0`` if this was not supplied)
+        The value of the `start` parameter(``0`` if this was not supplied).
```
Comment (datapythonista) [inline pandas/core/indexes/range.py]: same

File: pandas/core/indexes/range.py [hunk 9]
```diff
@@ -282,15 +282,15 @@ def _stop(self):
     @cache_readonly
     def step(self):
         """
-        The value of the `step` parameter (``1`` if this was not supplied)
+        The value of the `step` parameter(``1`` if this was not supplied).
```
Comment (datapythonista) [inline pandas/core/indexes/range.py]: same

File: pandas/core/indexes/range.py [hunk 10]
```diff
@@ -282,15 +282,15 @@ def _stop(self):
...
         """
         # GH 25710
         return self._range.step
 
     @property
     def _step(self):
         """
-        The value of the `step` parameter (``1`` if this was not supplied)
+        The value of the `step` parameter(``1`` if this was not supplied).
```
Comment (datapythonista) [inline pandas/core/indexes/range.py]: same

File: pandas/core/reshape/merge.py [hunk 11]
```diff
@@ -177,8 +177,8 @@ def merge_ordered(
 ):
     """
     Perform merge with optional filling/interpolation designed for ordered
-    data like time series data. Optionally perform group-wise merge (see
-    examples)
+    data like time series data. Optionally perform group-wise merge(see
```
Comment (datapythonista) [inline pandas/core/reshape/merge.py]: same

**Review by datapythonista — COMMENTED** [review]
Couple of small things, but almost there.

It may be useful to do a code review yourself once you're done, so you know exactly what you're proposing.

Thanks!

File: pandas/core/indexes/numeric.py (hunk 6 above)
Comment (bhuvanakundumani) [reply pandas/core/indexes/numeric.py]: Done.

**Review by datapythonista — APPROVED** [review]
Looks perfect now, thanks a lot for the fixes @bhuvanakundumani.

**Comment by WillAyd** [comment]
Thanks @bhuvanakundumani

**Comment by bhuvanakundumani** [comment]
Thanks @datapythonista @WillAyd.

**Pull request merged**